src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -1051,6 +1124,73 @@ def method(
 
         return res
 
+    def to_numpy(self, dtype: str) -> "ReplicatedSharedTensor":
+        """Converts the underlying tensor shares to numpy array.
+
+        Args:
+            dtype (str) : The data type to convert the tensor to.
+
+        Returns:
+            ReplicatedSharedTensor: converted RSTensor object.
+        """
+        rst = self.clone()
+        for idx, share in enumerate(self.shares):
+            rst.shares[idx] = share.numpy().astype(dtype)
+
+        return rst
+
+    def from_numpy(self) -> "ReplicatedSharedTensor":
+        """Converts the underlying tensor to torch tensor.
+
+        Returns:
+            ReplicatedSharedTensor: converted RSTensor object.
+        """
+        shares = copy.deepcopy(self.shares)
+        dtype = str(self.shares[0].dtype)
+        # Convert unsigned to signed as torch supports only signed.
+        dtype = SIGNED_MAP.get(dtype, dtype)
+        rst = ReplicatedSharedTensor(
+            session_uuid=self.session_uuid, config=self.config, ring_size=self.ring_size
+        )
+        rst.shares = []
+        for idx, share in enumerate(shares):
+            rst.shares.append(torch.from_numpy(share.astype(dtype)))
+
+        return rst
+
+    def wrap_rst(self, other) -> "ReplicatedSharedTensor":
+        """Applies wrap2 on the shares.
+
+        Args:
+            other (ReplicatedSharedTensor): tensor to wrap on.
+
+        Returns:
+            result (ReplicatedSharedTensor): Wrap of the tensor in binary.
+
+        Raises:
+            ValueError: If the input tensors are not numpy array.
+        """
+        from sympc.protocol import Falcon
+
+        if not (
+            isinstance(self.shares[0], np.ndarray)
+            and isinstance(other.shares[0], np.ndarray)
+        ):
+            raise ValueError("Input tensor to wrap should be a numpy array.")
+        shares = []
+
+        for x_share, y_share in zip(self.shares, other.shares):
+            shares.append(Falcon.wrap2(x_share, y_share))
+
+        config = Config(encoder_base=1, encoder_precision=0)
+        result = ReplicatedSharedTensor(
+            ring_size=2,
```

Comment:
Hmmmmm, Why ring size 2?

Reply:
The `Falcon.wrap2` returns an boolean numpy wrap(1 accounting for wrap , 0 accounting for no wrap)